modules/openid-federation-common/src/jvmTest/kotlin/com/sphereon/oid/fed/common/httpclient/OidFederationClientTest.kt

```diff
@@ -0,0 +1,64 @@
+package com.sphereon.oid.fed.common.httpclient
+
+import com.sphereon.oid.fed.openapi.models.*
+import io.ktor.client.engine.mock.*
+import io.ktor.http.*
+import kotlinx.coroutines.runBlocking
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import kotlin.test.Test
+
+class OidFederationClientTest {
+
+    private val entityStatement = EntityStatement(
+            iss = "https://edugain.org/federation",
+            sub = "https://openid.sunet.se",
+            exp = 1568397247,
+            iat = 1568310847,
+            sourceEndpoint = "https://edugain.org/federation/federation_fetch_endpoint",
+            jwks = JWKS(
+                propertyKeys = listOf(
+                    JWK(
+                        // missing e and n ?
+                        kid = "dEEtRjlzY3djcENuT01wOGxrZlkxb3RIQVJlMTY0...",
+                        kty = "RSA"
+                    )
+                )
+            ),
+            metadata = Metadata(
+                federationEntity = FederationEntityMetadata(
+                    organizationName = "SUNET"
+                )
+            )
+    )
+
+    private val mockEngine = MockEngine {
+        respond(
+            content = Json.encodeToString(entityStatement),
+            status = HttpStatusCode.OK,
+            headers = headersOf(HttpHeaders.ContentType, "application/entity-statement+jwt")
+        )
+    }
+
+    @Test
+    fun testGetEntityStatement() {
+        runBlocking {
+            val client = OidFederationClient(mockEngine)
+            val response = client.fetchEntityStatement("https://www.example.com?iss=https://edugain.org/federation&sub=https://openid.sunet.se", HttpMethod.Get)
+            assert(response == entityStatement)
```

Comment:
Kotlin test has an asser Equals right, instead of using the default built in assert.
When it fails this does not provide enough info

Reply:
Refactore to use assertEquals()